test/utils/helperFn.ts

```diff
@@ -55,13 +55,15 @@ export async function sleep(duration: number) {
 }
 
 export function getColumnValues(colIndex: number = 0) {
-  return Array.from(document.querySelectorAll(`[role="cell"][aria-colindex="${colIndex}"]`)).map(
-    (node) => node!.textContent,
-  );
+  return Array.from(
+    document.querySelectorAll(`[role="cell"][aria-colindex="${colIndex + 1}"]`),
+  ).map((node) => node!.textContent);
 }
 
 export function getColumnHeaderCell(colIndex: number): HTMLElement {
-  const columnHeader = document.querySelector(`[role="columnheader"][aria-colindex="${colIndex}"]`);
```

Comment:
I think this getter was incorrect as `aria-colindex` was `colIndex + 1`:
https://github.com/mui-org/material-ui-x/blob/5667025a4c0b5beb6cbf65141480daa9ff931a48/packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx#L195

so had to refactor the tests.